I1283 and i1324 cleaning creating repairing devfile section (#841)
codenvy/che-docs: src/main/pages/che-7/end-user-guide/con_what-is-a-devfile.adoc

// making-a-workspace-portable-using-a-devfile

[id="what-is-a-devfile_{context}"]
= What is a devfile

A devfile is a file that describes a development environment:

* the source code
* the development components (IDE tools and application runtimes)
* a list of pre-defined commands

> Devfiles are yaml files that Che consumes and transforms into a cloud workspace composed of multiple containers. The devfile can be saved in the root folder of the Git repository, a feature branch of a Git repository, a publicly accessible destination, or as a separate locally-stored artifact.

Reviewer: ```suggestion
Devfiles are YAML files that Che consumes and transforms into a cloud workspace composed of multiple containers. The devfile can be saved in the root folder of a Git repository, a feature branch of a Git repository, a publicly accessible destination, or as a separate, locally stored artifact.
```

> Che uses 'devfile' to define:

Reviewer: ```suggestion
Che uses a _devfile_ to define:
```

* projects to clone
* browser IDE to use
* preconfigured commands
* tools that you need

> * application runtime definition

Reviewer: This seems to (partially) duplicate the content above (lines 6--10).

> When you create a workspace, Che uses that definition to initiate everything for you and run all the containers for your tools and your application runtimes. Che also mounts filesystem volumes to make your source code available to the workspace.

Reviewer: ```suggestion
When creating a workspace, Che uses that definition to initiate everything and run all the containers for the required tools and application runtimes. Che also mounts file-system volumes to make source code available to the workspace.
```

> Devfiles can be versioned with the project source code. When you have to get a workspace to fix an old maintenance branch, the project devfile provides you with a definition of the workspace with the tools and the exact dependencies to start working on the old branch. Use it to easily instantiate workspaces on demand.

Reviewer: ```suggestion
Devfiles can be versioned with the project source code. When there is a need for a workspace to fix an old maintenance branch, the project devfile provides a definition of the workspace with the tools and the exact dependencies to start working on the old branch. Use it to instantiate workspaces on demand.
```

> Che maintains the devfile up-to-date with the tools you use in the workspace:

Reviewer: ```suggestion
Che maintains the devfile up-to-date with the tools used in the workspace:
```

* Projects of the workspace (path, Git location, branch)
* Commands to perform daily tasks (build, run, test, debug)

> * Runtime environment (container images to run your application)

Reviewer: ```suggestion
* Runtime environment (container images to run the application)
```
